src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final int FIRST_ELEMENT_OF_ARRAY = 0;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (isFull()) {
+            resizeArray();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < FIRST_ELEMENT_OF_ARRAY) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Index " + index + " is incorrect");
+        }
+        if (isFull()) {
+            resizeArray();
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            System.arraycopy(elements, index, elements, index + 1, size - index);
+            elements[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < FIRST_ELEMENT_OF_ARRAY) {
```

Comment:
Create a new method for checking the index. Fix in all places
[See here](https://github.com/mate-academy/jv-array-list/pull/741#pullrequestreview-1223384788)